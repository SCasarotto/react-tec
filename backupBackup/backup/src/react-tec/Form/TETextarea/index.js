import TETextarea from './TETextarea';

export default TETextarea;
