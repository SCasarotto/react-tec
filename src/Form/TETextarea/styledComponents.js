import styled from 'styled-components'

export const Textarea = styled.textarea`
	${(props) => {
		const { disabled, theme } = props

		return `
            position: relative;
            display: block;
            width: 100%;
            font-size: 14px;
            box-sizing: border-box;
            padding: 10px;
            background-color: ${theme.white};
            min-height: 140px;
            border-radius: 5px;
            line-height: 1.5;
            cursor: ${disabled ? 'not-allowed' : 'text'};

            color: ${disabled ? theme.gray : theme.darkerGray};
            border: 1px solid ${disabled ? theme.lighterGray : theme.lightGray};

            transition: border-color 0.2s ease-in, box-shadow 0.2s ease-in;
            :hover {
                border: 1px solid ${disabled ? theme.lighterGray : theme.primary};
                box-shadow: 0 0 0 1px ${disabled ? theme.lighterGray : theme.primary} inset;
           }
            :active {
                border: 1px solid ${disabled ? theme.lighterGray : theme.primary};
                box-shadow: 0 0 0 1px ${disabled ? theme.lighterGray : theme.primary} inset;
            }
        `
	}}
`
