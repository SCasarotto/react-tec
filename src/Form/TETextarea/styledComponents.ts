import styled from 'styled-components'

export const Textarea = styled.textarea`
	position: relative;
	display: block;
	width: 100%;
	font-size: 14px;
	box-sizing: border-box;
	padding: 10px;
	background-color: ${(props) => props.theme.white};
	min-height: 140px;
	border-radius: 5px;
	line-height: 1.5;

	color: ${(props) => props.theme.darkerGray};
	border: 1px solid ${(props) => props.theme.lightGray};

	transition: border-color 0.2s ease-in, box-shadow 0.2s ease-in;
	:hover,
	:active {
		border: 1px solid ${(props) => props.theme.primary};
		box-shadow: 0 0 0 1px ${(props) => props.theme.primary} inset;
	}

	:disabled {
		color: ${(props) => props.theme.gray};
		border: 1px solid ${(props) => props.theme.lighterGray};
		box-shadow: none;
		cursor: not-allowed;
	}
`
