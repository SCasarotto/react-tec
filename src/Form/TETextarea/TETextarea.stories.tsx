import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TETextarea, TETextareaProps } from './TETextarea'

export default {
	title: 'Form/TETextarea',
	component: TETextarea,
	argTypes: {},
} as Meta

const Template: Story<TETextareaProps> = (args) => <TETextarea {...args} />

export const Default = Template.bind({})
Default.args = {
	onChange: (e) => console.log(e.target.value),
}
