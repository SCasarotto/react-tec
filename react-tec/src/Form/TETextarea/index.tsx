import React from 'react'

import { Textarea } from './styledComponents'

export interface TETextareaProps
	extends React.DetailedHTMLProps<
		React.TextareaHTMLAttributes<HTMLTextAreaElement>,
		HTMLTextAreaElement
	> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
const TETextarea: React.FC<TETextareaProps> = (props) => {
	const { className = '', ...rest } = props
	return <Textarea className={`TETextarea ${className}`} {...rest} />
}

export default TETextarea
