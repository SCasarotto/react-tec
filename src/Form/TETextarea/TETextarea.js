import React from 'react'
import PropTypes from 'prop-types'

import { Textarea } from './styledComponents'

const TETextarea = (props) => {
	const { className = '', ...rest } = props
	return <Textarea className={`TETextarea ${className}`} {...rest} />
}

TETextarea.propTypes = {
	style: PropTypes.object,
	type: PropTypes.string,
	value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	onChange: PropTypes.func.isRequired,
	disabled: PropTypes.bool,
}

export default TETextarea
