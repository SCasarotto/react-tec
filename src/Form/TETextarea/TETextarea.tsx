import React from 'react'

import { Textarea } from './styledComponents'

export interface TETextareaProps extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}
/**
 *
 * TETextarea is a simple styled `<textarea>`.
 *
 * All props are passed right through to the input. Reference the [HTML documentation for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea).
 *
 */
export const TETextarea = React.forwardRef<HTMLTextAreaElement, TETextareaProps>((props, ref) => {
	const { className = '', ...rest } = props
	return <Textarea className={`TETextarea ${className}`} ref={ref} {...rest} />
})
