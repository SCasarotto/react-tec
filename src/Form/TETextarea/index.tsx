import React from 'react'

import { Textarea } from './styledComponents'

export interface TETextareaProps
	extends React.DetailedHTMLProps<
		React.TextareaHTMLAttributes<HTMLTextAreaElement>,
		HTMLTextAreaElement
	> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
export const TETextarea: React.FC<TETextareaProps> = (props) => {
	const { className = '', ...rest } = props
	return <Textarea className={`TETextarea ${className}`} {...rest} />
}
