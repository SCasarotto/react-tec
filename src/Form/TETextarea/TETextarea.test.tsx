import React from 'react'
import { render } from '@testing-library/react'
import { TETextarea } from './TETextarea'

describe('TETextarea', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TETextarea />)
		expect(asFragment()).toMatchSnapshot()
	})
})
