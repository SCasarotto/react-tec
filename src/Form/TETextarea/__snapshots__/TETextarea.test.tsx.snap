// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TETextarea matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 14px;
  box-sizing: border-box;
  padding: 10px;
  min-height: 140px;
  border-radius: 5px;
  line-height: 1.5;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
  transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
}

.c0:hover,
.c0:active {
  border: 1px solid;
  box-shadow: 0 0 0 1px inset;
}

.c0:disabled {
  border: 1px solid;
  box-shadow: none;
  cursor: not-allowed;
}

<textarea
    class="c0 TETextarea "
  />
</DocumentFragment>
`;
